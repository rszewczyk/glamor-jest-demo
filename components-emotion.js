import React from 'react'
import styled from 'react-emotion'

export const P = styled.p`
  color: ${props => (props.red ? 'red' : null)};
`
