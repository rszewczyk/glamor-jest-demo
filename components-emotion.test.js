import React from 'react'
import renderer from 'react-test-renderer'
import { P } from './components-emotion'
import serializer from 'jest-glamor-react'
import { sheet } from 'emotion'

expect.addSnapshotSerializer(
  serializer(sheet, (cn, index) => `emotion-${index}`)
)

it('renders as red', () => {
  expect(renderer.create(<P red />).toJSON()).toMatchSnapshot()
})

it('renders without red', () => {
  expect(renderer.create(<P />).toJSON()).toMatchSnapshot()
})
