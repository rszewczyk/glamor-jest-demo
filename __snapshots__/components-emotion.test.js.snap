// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as red 1`] = `
.emotion-0 {
  color: red;
}

<p
  className="emotion-0 emotion-1"
/>
`;

exports[`renders without red 1`] = `
<p
  className="css-olgnlr css-lnhyjl0"
/>
`;
